### ISSUE TITLE
Support CSS modules in imported npm packages

### ISSUE THREAD

**Issue by bjankord** [CompuIves/codesandbox-client#849]
# 🌈 Feature

# What is this feature?
I'd like to be able to run SCSS files / CSS files through CSS modules when included in an imported npm package if the file ends in `.module.scss` or `.module.css` similar to a feature @ro-savage added to create-react-app v2 branch.

## Opt-in only
The feature would be entirely optional and a user would have to explicitly opt in by using the naming convention and importing the css file into a JS file.

## Gotchas
If someone named a stylesheet [something].modules.css without it actually being a CSS Module and imported it, it would be imported as a CSS Module.

## How things work currently
I've created a [react button component](https://github.com/bjankord/test-react-ui-button) that [imports a SCSS file](https://github.com/bjankord/test-react-ui-button/blob/master/src/Button.jsx#L5) that should be run through CSS modules. However, when the component is imported into a codesandbox, the styles are not run through CSS modules.

I've also created some sandboxes with various templates that use this button component:
[Parcel Template](https://codesandbox.io/s/003zx4v0on) - Styles from SCSS file load but are not run through CSS modules
[Preact Template](https://codesandbox.io/s/4wn0zx3q59) - Styles from SCSS file load but are not run through CSS modules
[Create React App / Webpack Template](https://codesandbox.io/s/v065225zz3) - Styles don't load, this is expected since codesandbox currently follows create-react-app v1 template which doesn't support SCSS.

```jsx
// current output
<button class="button neutral" type="button" aria-disabled="false">
  <span class="text-only">
    <span class="">hello</span>
  </span>
</button>
```

```jsx
// desired output
<button class="_button_2jjfa_1 _neutral_1ffga_2" type="button" aria-disabled="false">
  <span class="_text-only_1jafa_1">
    <span class="">hello</span>
  </span>
</button>
```

I'm not too concerned about the excat module hasing syntax, here is what [create-react-app-next used originally](https://github.com/facebook/create-react-app/pull/2285/files#diff-a7f98c18479be87c9f33e7604dbd1a09R255), however they [updated it recently](https://github.com/facebook/create-react-app/blob/32eebfeb7f5cdf93a790318eb76b81bb2927458e/packages/react-dev-utils/getCSSModuleLocalIdent.js).

Ideally, I'm looking for this feature to be added to the parcel template. I just made demos with all 3  templates to see how the component / styles worked in each.

## Other projects that support this feature
In the create-react-app v2 branch, they've added support for this feature where [.module.css](https://github.com/facebook/create-react-app/pull/2285) and [.module.scss](https://github.com/facebook/create-react-app/pull/4195/files) are automatically run through CSS modules via [regex check on file names](https://github.com/facebook/create-react-app/blob/next/packages/react-scripts/config/webpack.config.dev.js#L51-L53).

[create-react-app-parcel](https://github.com/sw-yx/create-react-app-parcel/blob/master/packages/react-scripts-parcel/config/parcel.config.prod.js#L44-L46) also supports this feature. 

[webpacker](https://github.com/rails/webpacker), a rails gem, also [supports this convention](https://github.com/rails/webpacker/pull/1248).

poi also [includes support](https://poi.js.org/guide/transforms.html#css-modules) for this feature.

# How the feature should work?
Style files that end in `.module.scss` or `.module.css` should be run through CSS modules automatically. This should work both for files in the local file directory and files imported from npm like the [sample react button component](https://github.com/bjankord/test-react-ui-button/blob/master/src/Button.jsx#L5) 

# You have examples or an idea how it can be implemented? 
I've looked through a [few](https://github.com/CompuIves/codesandbox-client/blob/580729dcfb455e04e0b622f2436d7387189f3545/packages/app/src/sandbox/eval/transpilers/sass/worker/sass-worker.js) [files](https://github.com/CompuIves/codesandbox-client/blob/580729dcfb455e04e0b622f2436d7387189f3545/packages/app/src/sandbox/eval/transpilers/babel/worker/plugins/babel-plugin-dynamic-css-modules.js) in codesandbox to see if I could understand how to implement this feature, but I wasn't really able to wrap my head around how to introduce this feature into the codebase.

**Comment by CompuIves** [comment]
I think this one shouldn't be too hard to add 😄. We can add a check for `.module.css` and automatically export css modules, this would also work on `node_modules`.

I think it would make sense to add this to the `create-react-app` template as soon as CRA v2 is officially released, to prevent confusion. So let's keep this one open until then!

**Comment by bjankord** [comment]
Sounds good, can you also add a check for `.module.scss` [similar to CRA v2](https://github.com/facebook/create-react-app/pull/4195/files#diff-a7f98c18479be87c9f33e7604dbd1a09R53) when it rolls out.

**Comment by JasonKleban** [comment]
Is there a way to use create-react-app v2 (especially the typescript one) in codesandbox currently?  I'm trying to use css-modules and in the current version I guess I can't?

**Comment by bjankord** [comment]
@JasonKleban I don't think there is a way to use react-scripts v2 yet, @CompuIves correct me if I'm wrong. create-react-app v2 is in final beta stage so it won't be much longer before react-scripts v2.0.0 stable is available. Hopefully, then, codesandbox will updated to support it.

**Comment by CompuIves** [comment]
Yep, that's right @bjankord. I will focus on putting Server Side Evaluation out first and then focus on CRA v2 right after 😄.

**Comment by bjankord** [comment]
@CompuIves I believe this can be closed out now 😄

**Comment by lbogdan** [comment]
Fixed in #1124 , closing.
